src/MLJCatBoostInterface.jl

```diff
@@ -72,12 +72,30 @@ end
 include("mlj_catboostclassifier.jl")
 include("mlj_catboostregressor.jl")
 
-const CATBOOST_MODELS = Union{CatBoostClassifier,CatBoostRegressor}
+const CatBoostModels = Union{CatBoostClassifier,CatBoostRegressor}
+
+function MMI.selectrows(::CatBoostModels, I, X, y)
+    py_I = numpy.array(numpy.array(I))
+    return py_X.iloc[py_I,], y[py_I]
+end
+
+function MMI.update(mlj_model::CatBoostModels, verbosity::Integer, fitresult, cache, X, y)
+    current_iterations = pyconvert(Int, mach.fitresult.tree_count_)
+    if current_iterations < mlj_model.iterations
+        iterations = mlj_model.iterations - current_iterations
+        fitresult.fit(X, y; init_model=fitresult, iterations=iterations)
+    else
```

Comment:
You should check that `iterations` is the *only* hyperparameter that has changed. To that end I suggest that `fit` (and `update`) makes a deepcopy of `model`, which it returns as `cache` (2nd return value), which MLJ will ensure gets passed as `cache` in the `update` call being implemented here.  You can then compare the current `model` with the last one used in any training, `cache` (and also have access to previous `iterations` without counting trees in the `fitresult`). MLJModelInterface has a convenience function, [`is_same_except`](https://github.com/JuliaAI/MLJModelInterface.jl/blob/d9e9703947fc04b0a5e63680289e41d0ba0d65bd/src/equality.jl#L74)  to make the comparison easy. You can see it applied [here](https://github.com/FluxML/MLJFlux.jl/blob/452c09d7dc13914f4057c661448bc310c9362d3d/src/mlj_model_interface.jl#L115).

Reply:
I am a little stumped on the best way to do this. `is_same_except` doesn't work with the python objects. I have updated `MMI.update` to create a brand new CatBoost model to start training where the last one left off. (It doesn't like resuming an already trained model, so I have to create a new one and initialize it with the previous one).

Reply:
> I am a little stumped on the best way to compare hyperparameters. is_same_except doesn't work with the python objects. There are a few differences between the two models, Should I compare the two parameter dictionaries?

I don't understand what python has to do with it. You are comparing two Julia structs, with ordinary fields no?

Okay, i just had a look at your [Julia structs](https://github.com/beacon-biosignals/CatBoost.jl/blob/45647592ed06c43d7267f5622c7fdc2399796ecc/src/mlj_catboostclassifier.jl#L2) for the model hyperparameters and see you are doing something very weird. See my other [comment](https://github.com/beacon-biosignals/CatBoost.jl/pull/16#discussion_r1082047909). MLJ cannot control an iterative model where the default value of the iteration parameter is `nothing`. It needs to have a value. And it cannot change its value except by passing an instance with the new value.

Reply:
Ah, I see. I updated it [here](https://github.com/beacon-biosignals/CatBoost.jl/pull/16/commits/3f961b7772e08ee204a7074230f0a3a3adadf2d9) to compare the mlj model and the cache version from the previous update.

Reply:
resolved